Update error message to accommodate OCI: "Registry may not support Image Manifest Version 2, Schema 2"

```
Registry may not support Image Manifest Version 2, Schema 2 
at com.google.cloud.tools.jib.registry.RegistryErrorExceptionBuilder.build(RegistryErrorExceptionBuilder.java:108)
	at com.google.cloud.tools.jib.registry.ManifestPusher.handleHttpResponseException(ManifestPusher.java:119)
	at com.google.cloud.tools.jib.registry.ManifestPusher.handleHttpResponseException(ManifestPusher.java:39)
	at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:258)
	at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.fallBackToHttp(RegistryEndpointCaller.java:210)
	at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.callWithAllowInsecureRegistryHandling(RegistryEndpointCaller.java:180)
	at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:161)
	at com.google.cloud.tools.jib.registry.RegistryClient.callRegistryEndpoint(RegistryClient.java:356)
	at com.google.cloud.tools.jib.registry.RegistryClient.pushManifest(RegistryClient.java:225)
	at com.google.cloud.tools.jib.builder.steps.PushImageStep.lambda$afterAllPushed$0(PushImageStep.java:162)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.google.api.client.http.HttpResponseException: 400 Bad Request
{"errors":[{"code":"MANIFEST_INVALID","message":"manifest invalid","detail":{}}]}

	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1102)
	at com.google.cloud.tools.jib.http.Connection.send(Connection.java:200)
	at com.google.cloud.tools.jib.registry.RegistryEndpointCaller.call(RegistryEndpointCaller.java:251)
	... 12 more
```

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/ManifestPusher.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/ManifestPusher.java
@@ -115,7 +115,9 @@ class ManifestPusher implements RegistryEndpointProvider<DescriptorDigest> {
     ErrorCodes errorCode = ErrorResponseUtil.getErrorCode(httpResponseException);
     if (errorCode == ErrorCodes.MANIFEST_INVALID || errorCode == ErrorCodes.TAG_INVALID) {
       throw new RegistryErrorExceptionBuilder(getActionDescription(), httpResponseException)
-          .addReason("Registry may not support Image Manifest Version 2, Schema 2")
+          .addReason(
+              "Registry may not support pushing OCI Manifest or "
+                  + "Docker Image Manifest Version 2, Schema 2")
           .build();
     }
     // rethrow: unhandled error response code.

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/ManifestPusherTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/ManifestPusherTest.java
@@ -163,7 +163,8 @@ public class ManifestPusherTest {
         new HttpResponseException.Builder(
                 HttpStatus.SC_BAD_REQUEST, "Bad Request", new HttpHeaders())
             .setContent(
-                "{\"errors\":[{\"code\":\"TAG_INVALID\",\"message\":\"manifest tag did not match URI\"}]}")
+                "{\"errors\":[{\"code\":\"TAG_INVALID\","
+                    + "\"message\":\"manifest tag did not match URI\"}]}")
             .build();
     try {
       testManifestPusher.handleHttpResponseException(exception);
@@ -173,7 +174,8 @@ public class ManifestPusherTest {
       Assert.assertThat(
           ex.getMessage(),
           CoreMatchers.containsString(
-              "Registry may not support Image Manifest Version 2, Schema 2"));
+              "Registry may not support pushing OCI Manifest or "
+                  + "Docker Image Manifest Version 2, Schema 2"));
     }
   }
 
@@ -185,7 +187,8 @@ public class ManifestPusherTest {
         new HttpResponseException.Builder(
                 HttpStatus.SC_BAD_REQUEST, "Bad Request", new HttpHeaders())
             .setContent(
-                "{\"errors\":[{\"code\":\"MANIFEST_INVALID\",\"message\":\"manifest invalid\",\"detail\":{}}]}")
+                "{\"errors\":[{\"code\":\"MANIFEST_INVALID\","
+                    + "\"message\":\"manifest invalid\",\"detail\":{}}]}")
             .build();
     try {
       testManifestPusher.handleHttpResponseException(exception);
@@ -195,7 +198,8 @@ public class ManifestPusherTest {
       Assert.assertThat(
           ex.getMessage(),
           CoreMatchers.containsString(
-              "Registry may not support Image Manifest Version 2, Schema 2"));
+              "Registry may not support pushing OCI Manifest or "
+                  + "Docker Image Manifest Version 2, Schema 2"));
     }
   }
 
@@ -206,8 +210,9 @@ public class ManifestPusherTest {
         new HttpResponseException.Builder(
                 HttpStatus.SC_UNSUPPORTED_MEDIA_TYPE, "UNSUPPORTED MEDIA TYPE", new HttpHeaders())
             .setContent(
-                "{\"errors\":[{\"code\":\"MANIFEST_INVALID\",\"detail\":"
-                    + "{\"message\":\"manifest schema version not supported\"},\"message\":\"manifest invalid\"}]}")
+                "{\"errors\":[{\"code\":\"MANIFEST_INVALID\","
+                    + "\"detail\":{\"message\":\"manifest schema version not supported\"},"
+                    + "\"message\":\"manifest invalid\"}]}")
             .build();
     try {
       testManifestPusher.handleHttpResponseException(exception);
@@ -217,7 +222,8 @@ public class ManifestPusherTest {
       Assert.assertThat(
           ex.getMessage(),
           CoreMatchers.containsString(
-              "Registry may not support Image Manifest Version 2, Schema 2"));
+              "Registry may not support pushing OCI Manifest or "
+                  + "Docker Image Manifest Version 2, Schema 2"));
     }
   }
 

